### PR TITLE
feat(examples): add prediction market demo realm

### PR DESCRIPTION
**Pull request by zardozmonopoly**
Adds a parimutuel-style prediction market realm under examples/gno.land/r/demo/predictionmarket.

- Uses chain/banker (BankerTypeRealmSend) for escrow and payout
- Uses unsafe.OriginSend() to read the attached bet amount
- Admin creates markets and resolves outcomes; users bet ugnot on Home/Away/Draw and claim winnings proportional to the losing pool
- Tests cover market creation, resolution, double-resolve guard, and early-claim guard

Happy to iterate on naming, structure, or add more edge-case tests based on feedback.

### CODE REVIEW

**Comment by Gno2D2** [comment]
#### 🛠 PR Checks Summary
 🔴 Pending initial approval by a review team member, or review from tech-staff


##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)
- [ ] The pull request description provides enough details


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🔴 Pending initial approval by a review team member, or review from tech-staff


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 The pull request was created from a fork (head branch repo: zardozmonopoly/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>Pending initial approval by a review team member, or review from tech-staff</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 Not (🔴 Pull request author is a member of the team: tech-staff)

```
### Then
```
🔴 Requirement not satisfied
└── 🔴 If
    ├── 🔴 Condition
    │   └── 🔴 Or
    │       ├── 🔴 At least one of these user(s) reviewed the pull request: [aronpark1007 davd-gzl jefft0 notJoon omarsy MikaelVallenet] (with state "APPROVED")
    │       ├── 🔴 At least 1 user(s) of the team tech-staff reviewed pull request
    │       └── 🔴 This pull request is a draft
    └── 🔴 Else
        └── 🔴 And
            ├── 🟢 This label is applied to pull request: review/triage-pending
            └── 🔴 On no pull request

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

<details><summary>The pull request description provides enough details</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 Not (🔴 Pull request author is a member of the team: core-contributors)
    └── 🟢 Not (🔴 Pull request author is user: dependabot[bot])

```
### Can be checked by
 - team core-contributors

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

**Comment by zardozmonopoly** [comment]
For context: I run a Test13 testnet validator (moniker: zardozmonopoly). Happy to help test any fix or provide more detail if useful..

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+func PlaceBet(cur realm, marketID int, outcome int) {
+	runtime.AssertOriginCall()
+	if !cur.Previous().IsUserCall() {
+		panic("only direct EOA calls can place bets")
+	}
+	if outcome < 0 || outcome > 2 {
+		panic("invalid outcome")
+	}
+
+	raw, exists := markets.Get(strconv.Itoa(marketID))
```
Comment (davd-gzl) [inline examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Critical: [`avl.Tree.Get`](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/p/nt/avl/v0/tree.gno#L58) returns a single value, so this assignment and the four others at lines [107](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/r/demo/predictionmarket/predictionmarket.gno#L107), [121](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/r/demo/predictionmarket/predictionmarket.gno#L121), [14](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/r/demo/predictionmarket/predictionmarket_test.gno#L14) and [31](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/r/demo/predictionmarket/predictionmarket_test.gno#L31) fail to typecheck. The four tests the description lists have never run, and [`ci / examples`](https://github.com/gnolang/gno/blob/037a90410/.github/workflows/ci-dir-examples.yml#L7-L10) will fail once it is released to run.

<details><summary>repro</summary>

```bash
# from a local clone of gnolang/gno:
gh pr checkout 5946 -R gnolang/gno
cd examples/gno.land/r/demo/predictionmarket
go run ../../../../../gnovm/cmd/gno test -v .
```

```
predictionmarket.gno:74:17: assignment mismatch: 2 variables but markets.Get returns 1 value (code=gnoTypeCheckError)
predictionmarket.gno:107:17: assignment mismatch: 2 variables but markets.Get returns 1 value (code=gnoTypeCheckError)
predictionmarket.gno:121:17: assignment mismatch: 2 variables but markets.Get returns 1 value (code=gnoTypeCheckError)
FAIL    . 	0.62s
FAIL
FAIL: 0 build errors, 1 test errors
```
</details>

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+	var totalPool int64
+	for _, p := range m.pools {
+		totalPool += p
+	}
+
+	var payout int64
+	for _, b := range m.bets {
+		if b.bettor == caller && b.outcome == m.winner && b.amount > 0 {
+			payout += (b.amount * totalPool) / winningPool
```
Comment (davd-gzl) [inline examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Critical: `b.amount * totalPool` wraps `int64` before the division, so the payout goes negative and the [banker send](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/r/demo/predictionmarket/predictionmarket.gno#L152) aborts. The limit is on the product, not on either amount, so 5,000,000,000 ugnot on each of two outcomes already crosses it and every bettor in that market is locked out for good. [`math/overflow`](https://github.com/gnolang/gno/blob/037a90410/gnovm/stdlibs/math/overflow/overflow_generated.gno#L583-L589) turns the wrap into a panic, as [`p/demo/tokens/grc20`](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/p/demo/tokens/grc20/token.gno#L193-L194) does.

<details><summary>repro</summary>

```bash
# from a local clone of gnolang/gno:
gh pr checkout 5946 -R gnolang/gno
D=examples/gno.land/r/demo/predictionmarket

# the package does not build yet; apply the avl.Get arity fix first
sed -i 's/raw, exists := markets.Get(\(.*\))/raw := markets.Get(\1)/; s/if !exists {/if raw == nil {/' $D/predictionmarket.gno
sed -i 's/raw, exists := markets.Get(\(.*\))/raw := markets.Get(\1)/; s/raw, _ := markets.Get(\(.*\))/raw := markets.Get(\1)/; s/if !exists {/if raw == nil {/' $D/predictionmarket_test.gno

cat > $D/payout_probe_test.gno <<'EOF'
package predictionmarket

import (
	"chain"
	"chain/banker"
	"strconv"
	"testing"
)

func TestClaimPayoutDoesNotWrap(cur realm, t *testing.T) {
	rlm := chain.PackageAddress("gno.land/r/demo/predictionmarket")
	testing.IssueCoins(rlm, chain.Coins{{"ugnot", 1_000_000_000_000}})
	winner := cur.Previous().Address()
	before := banker.NewReadonlyBanker().GetCoins(winner).AmountOf("ugnot")

	// 5000 GNOT on each side: the sole Home bettor takes the whole 10000 GNOT.
	const stake = int64(5_000_000_000)
	id := CreateMarket(cur, "overflow probe")
	m := markets.Get(strconv.Itoa(id)).(*Market)
	m.pools[OutcomeHome] = stake
	m.pools[OutcomeAway] = stake
	m.bets = append(m.bets, &Bet{bettor: winner, outcome: OutcomeHome, amount: stake})
	m.resolved = true
	m.winner = OutcomeHome
	ClaimWinnings(cur, id)

	got := banker.NewReadonlyBanker().GetCoins(winner).AmountOf("ugnot") - before
	if got != 2*stake {
		t.Fatalf("payout: got %d, want %d", got, 2*stake)
	}
}
EOF

cd $D && go run ../../../../../gnovm/cmd/gno test -v -run TestClaimPayoutDoesNotWrap . 2>&1 | head -3
cd - && rm $D/payout_probe_test.gno && git checkout $D
```

```
=== RUN   TestClaimPayoutDoesNotWrap
panic: invalid result:  + -1068046444ugnot = -1068046444ugnot: non-positive coin amount: -1068046444
```
</details>

**Review by davd-gzl — COMMENTED** [review]
[AI bot - Automatic review]

Automated technical pass: does the code build, run, and behave as described. No design or scope judgement, and no merge verdict. Posted to give a human reviewer a head start.

Verified on 037a90410: with the `avl.Get` arity fixed so the package builds, a claim at 5,000,000,000 ugnot per side panics inside the banker send, and succeeds once the division comes first. Go returns the same wrapped values, so this is `int64` semantics, not a GnoVM divergence.

[`ClaimWinnings`](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/r/demo/predictionmarket/predictionmarket.gno#L119-L153) is the only path that moves coins out of the realm, so any case it refuses strands the escrow for good.

Realms under `examples/` are [pre-deployed to gno.land testnets](https://github.com/gnolang/gno/blob/037a90410/examples/README.md?plain=1#L7-L9), so this holds real testnet ugnot once merged.

Full review: https://github.com/samouraiworld/gno-agent-workspace/blob/main/reviews/pr/5xxx/5946-prediction-market-demo-realm/1-037a90410/review_claude-opus-4-8_davd-gzl.md

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	}
+	m := raw.(*Market)
+	if !m.resolved {
+		panic("market not resolved yet")
+	}
+
+	winningPool := m.pools[m.winner]
+	if winningPool == 0 {
+		panic("no winning pool")
+	}
```
Comment (davd-gzl) [inline examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Critical: when the declared outcome drew no bets, every claim panics here and the escrow stays in the realm for good. [`ResolveMarket`](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/r/demo/predictionmarket/predictionmarket.gno#L104-L106) accepts any outcome in 0..2 without checking that anyone backed it, so a draw between two bettors split across Home and Away hits this on the first try. A market the admin never resolves ends the same way.

<details><summary>repro</summary>

```bash
# from a local clone of gnolang/gno:
gh pr checkout 5946 -R gnolang/gno
D=examples/gno.land/r/demo/predictionmarket

# the package does not build yet; apply the avl.Get arity fix first
sed -i 's/raw, exists := markets.Get(\(.*\))/raw := markets.Get(\1)/; s/if !exists {/if raw == nil {/' $D/predictionmarket.gno
sed -i 's/raw, exists := markets.Get(\(.*\))/raw := markets.Get(\1)/; s/raw, _ := markets.Get(\(.*\))/raw := markets.Get(\1)/; s/if !exists {/if raw == nil {/' $D/predictionmarket_test.gno

cat > $D/empty_pool_probe_test.gno <<'EOF'
package predictionmarket

import (
	"strconv"
	"testing"
)

func TestClaimWithEmptyWinningPool(cur realm, t *testing.T) {
	bettor := cur.Previous().Address()
	id := CreateMarket(cur, "empty pool probe")
	m := markets.Get(strconv.Itoa(id)).(*Market)
	m.pools[OutcomeHome] = 1_000_000
	m.pools[OutcomeAway] = 1_000_000
	m.bets = append(m.bets, &Bet{bettor: bettor, outcome: OutcomeHome, amount: 1_000_000})
	m.resolved = true
	m.winner = OutcomeDraw // nobody backed Draw

	defer func() { t.Fatalf("claim panicked: %v; the 2000000ugnot escrow has no exit", recover()) }()
	ClaimWinnings(cur, id)
}
EOF

cd $D && go run ../../../../../gnovm/cmd/gno test -v -run TestClaimWithEmptyWinningPool . 2>&1 | grep -E 'RUN|claim panicked|FAIL'
cd - && rm $D/empty_pool_probe_test.gno && git checkout $D
```

```
=== RUN   TestClaimWithEmptyWinningPool
claim panicked: no winning pool; the 2000000ugnot escrow has no exit
--- FAIL: TestClaimWithEmptyWinningPool (0.00s)
FAIL    . 	1.22s
FAIL
```
</details>

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+	caller := cur.Previous().Address()
+	m.pools[outcome] += amount
+	m.bets = append(m.bets, &Bet{bettor: caller, outcome: outcome, amount: amount})
+}
+
+func ResolveMarket(cur realm, marketID int, winningOutcome int) {
+	caller := cur.Previous().Address()
+	if caller != admin {
+		panic("only admin can resolve markets")
+	}
```
Comment (davd-gzl) [inline examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Nothing stops the admin from betting, then resolving their own outcome and taking the pot: [`PlaceBet`](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/r/demo/predictionmarket/predictionmarket.gno#L65-L72) admits the admin like any account, and there is no betting cutoff, oracle or dispute window. The package carries no doc comment saying a bettor is trusting the admin completely.

File: examples/gno.land/r/demo/predictionmarket/predictionmarket_test.gno [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+func TestClaimWinningsBeforeResolveFails(cur realm, t *testing.T) {
+	id := CreateMarket(cur, "Test Market: Early Claim")
+
+	defer func() {
+		if r := recover(); r == nil {
+			t.Fatal("expected panic on claim before resolve")
+		}
+	}()
+	ClaimWinnings(cur, id)
+}
```
Comment (davd-gzl) [inline examples/gno.land/r/demo/predictionmarket/predictionmarket_test.gno:141]: Missing test: nothing places a bet or completes a claim, so both the overflow and the empty-winning-pool lock reach the chain unchallenged. The admin gate is also vacuous under `gno test`, where `unsafe.OriginCaller()` and `cur.Previous().Address()` are both the empty address.

<details><summary>test cases</summary>

`PlaceBet` cannot be driven from an in-package test, because [`AssertOriginCall`](https://github.com/gnolang/gno/blob/037a90410/gnovm/stdlibs/chain/runtime/native.go#L9-L27) requires the first frame to be a message call; a txtar integration test is the right home for the bet path. The payout assertions seed the market directly. Red at 037a90410 with the arity fix applied, green once the multiplication stays in range.

```go
package predictionmarket

import (
	"chain"
	"chain/banker"
	"strconv"
	"testing"
)

const realmPath = "gno.land/r/demo/predictionmarket"

// seedResolved builds a resolved market directly, bypassing PlaceBet's
// origin-call guard, which no in-package test can satisfy.
func seedResolved(cur realm, bettor address, stake int64) int {
	id := CreateMarket(cur, "overflow probe")
	m := markets.Get(strconv.Itoa(id)).(*Market)
	m.pools[OutcomeHome] = stake
	m.pools[OutcomeAway] = stake
	m.bets = append(m.bets, &Bet{bettor: bettor, outcome: OutcomeHome, amount: stake})
	m.resolved = true
	m.winner = OutcomeHome
	return id
}

func TestClaimPayoutDoesNotWrap(cur realm, t *testing.T) {
	testing.IssueCoins(chain.PackageAddress(realmPath), chain.Coins{{"ugnot", 1_000_000_000_000}})
	winner := cur.Previous().Address()
	before := banker.NewReadonlyBanker().GetCoins(winner).AmountOf("ugnot")

	// 5000 GNOT on each side: the sole Home bettor takes the whole 10000 GNOT.
	const stake = int64(5_000_000_000)
	ClaimWinnings(cur, seedResolved(cur, winner, stake))

	got := banker.NewReadonlyBanker().GetCoins(winner).AmountOf("ugnot") - before
	if got != 2*stake {
		t.Fatalf("payout: got %d, want %d", got, 2*stake)
	}
}

func TestClaimPayoutSmallStakeUnchanged(cur realm, t *testing.T) {
	testing.IssueCoins(chain.PackageAddress(realmPath), chain.Coins{{"ugnot", 1_000_000_000_000}})
	winner := cur.Previous().Address()
	before := banker.NewReadonlyBanker().GetCoins(winner).AmountOf("ugnot")

	const stake = int64(1_000_000)
	ClaimWinnings(cur, seedResolved(cur, winner, stake))

	got := banker.NewReadonlyBanker().GetCoins(winner).AmountOf("ugnot") - before
	if got != 2*stake {
		t.Fatalf("payout: got %d, want %d", got, 2*stake)
	}
}
```
</details>

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+	if payout == 0 {
+		panic("nothing to claim")
+	}
+
+	bkr := banker.NewBanker(banker.BankerTypeRealmSend, cur)
+	bkr.SendCoins(cur.Address(), caller, chain.Coins{{Denom: betDenom, Amount: payout}})
+}
+
+func Render(path string) string {
```
Comment (davd-gzl) [inline examples/gno.land/r/demo/predictionmarket/predictionmarket.gno:219]: Nit: `Render` ignores its `path` argument, so a single market page, a bettor's open positions and a claimable balance are all unreachable from the web view.

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+		panic("only admin can create markets")
+	}
+	id := nextID
+	nextID++
+	m := &Market{
+		id:          id,
+		description: description,
+		winner:      -1,
+	}
+	markets.Set(strconv.Itoa(id), m)
```
Comment (davd-gzl) [inline examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Nit: the tree key is the decimal id and [`Iterate`](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/p/nt/avl/v0/tree.gno#L89) walks keys as strings, so `Render` lists market 10 before market 2.

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+var (
+	markets avl.Tree
+	nextID  int
+	admin   address
+)
+
+func init() {
+	admin = unsafe.OriginCaller()
+	nextID = 1
+}
```
Comment (davd-gzl) [inline examples/gno.land/r/demo/predictionmarket/predictionmarket.gno:57]: Suggestion: `admin` is fixed at deploy with no transfer path, so a lost key leaves every open market unresolvable and its escrow stranded. [`p/nt/ownable/v0`](https://github.com/gnolang/gno/blob/037a90410/examples/gno.land/p/nt/ownable/v0/ownable.gno#L1) is the usual shape in `examples/`.

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+
+func PlaceBet(cur realm, marketID int, outcome int) {
+	runtime.AssertOriginCall()
+	if !cur.Previous().IsUserCall() {
+		panic("only direct EOA calls can place bets")
+	}
+	if outcome < 0 || outcome > 2 {
+		panic("invalid outcome")
+	}
```
Comment (davd-gzl) [inline examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Suggestion: the accepted outcomes are hardcoded here, so a market cannot declare itself binary and a bettor can stake on Draw in a contest that cannot draw.

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno (hunk 1 above)
Comment (zardozmonopoly) [reply examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Yeah, that's a real one -- fixed, and confirmed gno test actually runs now (it hadn't been before)..

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno (hunk 2 above)
Comment (zardozmonopoly) [reply examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Good catch. Same overflow.Mul64p fix as the other two PRs..

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno (hunk 3 above)
Comment (zardozmonopoly) [reply examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Fixed the no-bets-on-resolved-outcome case with a refund path. Didn't add anything for a market that's never resolved at all, though -- that would need some kind of deadline/timeout mechanism, which felt like a bigger scope change than this fix pass. Open to adding it if that's the preferred direction..

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno (hunk 4 above)
Comment (zardozmonopoly) [reply examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Fair -- there's no oracle here so the trust model is what it is, but at minimum the admin shouldn't be able to bet in their own market. Added that guard and said so directly in the doc comment instead of leaving it implicit..

File: examples/gno.land/r/demo/predictionmarket/predictionmarket_test.gno (hunk 5 above)
Comment (zardozmonopoly) [reply examples/gno.land/r/demo/predictionmarket/predictionmarket_test.gno:141]: Added real end-to-end tests (bet -> resolve -> claim) plus the overflow and no-winner-refund cases specifically. Also fixed the admin gate being vacuous under gno test -- tests now set up a real admin address via ownable.NewWithAddress and testing.SetOriginCaller instead of relying on the empty address unsafe.OriginCaller() returns in the test environment..

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno (hunk 6 above)
Comment (zardozmonopoly) [reply examples/gno.land/r/demo/predictionmarket/predictionmarket.gno:219]: Added a ?id= query for a single market's page. Didn't add a per-bettor positions or claimable-balance view yet -- that'd need a way to look up a market's bets by address, which the current Bet slice doesn't index for. Happy to add if useful..

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno (hunk 7 above)
Comment (zardozmonopoly) [reply examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Zero-padded the keys so this sorts numerically now..

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno (hunk 8 above)
Comment (zardozmonopoly) [reply examples/gno.land/r/demo/predictionmarket/predictionmarket.gno:57]: Switched to p/nt/ownable/v0 for this, so ownership can actually move if a key is lost..

File: examples/gno.land/r/demo/predictionmarket/predictionmarket.gno (hunk 9 above)
Comment (zardozmonopoly) [reply examples/gno.land/r/demo/predictionmarket/predictionmarket.gno]: Added an allowDraw flag per market so a market that can't end in a draw doesn't accept bets on one..
